3rdparty/dlmalloc/allocator.c

```diff
@@ -0,0 +1,117 @@
+// Copyright (c) Open Enclave SDK contributors.
+// Licensed under the MIT License.
+
+#include <openenclave/internal/allocator.h>
+
+#define HAVE_MMAP 0
+#define LACKS_UNISTD_H
+#define LACKS_SYS_PARAM_H
+#define LACKS_SYS_TYPES_H
+#define LACKS_TIME_H
+#define MORECORE dlmalloc_sbrk
+#define ABORT oe_abort()
+#define USE_DL_PREFIX
+#define LACKS_STDLIB_H
+#define LACKS_STRING_H
+#define USE_LOCKS 1
+#define fprintf _dlmalloc_stats_fprintf
+#define NO_MALLOC_STATS 1
+
+#pragma GCC diagnostic push
+#ifdef __clang__
+#pragma GCC diagnostic ignored "-Wparentheses-equality"
+#endif
+
+OE_INLINE
+int sched_yield(void)
+{
+#ifdef __x86__
+    asm volatile("pause");
```

Comment:
On non-x86 that means this function is broken right? Do OP-TEE enclaves use this allocator as well? The `yield` instruction on arm should do the same thing.

Reply:
This is the current behavior. When dlmalloc is compiled for OPTEE, sched yield is empty:
https://github.com/openenclave/openenclave/blob/71edd2cf92f52017410d5e3262628b962067560a/enclave/core/optee/sched_yield.c#L4-L7

@HernanGatta Do you know what the equivalent of `pause` assembly instruction is for OPTEE/ARM?

Reply:
Since OPTEE is currently `single-threaded` and since no-op is the current behavior, I'm resolving this for now.